test/extended/authentication.sh

```diff
@@ -65,28 +65,65 @@ wait_for_command 'oc get imagestream openldap --template="{{with \$tags := .stat
 # kick off a build and wait for it to finish
 oc start-build openldap --follow
 
-# wait for the deployment to be up and running
-wait_for_command 'oc get pods -l deploymentconfig=openldap-server --template="{{with \$items := .items}}{{with \$item := index \$items 0}}{{\$item.status.phase}}{{end}}{{end}}" | grep Running' $((60*TIME_SEC))
+# wait for LDAP server to be ready
+wait_for_command 'oc get pods -l deploymentconfig=openldap-server --template="{{with \$items := .items}}{{with \$item := index \$items 0}}{{with \$map := index \$item.status.conditions 0}}{{index \$map \"type\"}}{{end}}{{end}}{{end}}" | grep Ready' $((60*TIME_SEC)) 
 
 LDAP_SERVICE_IP=$(oc get --output-version=v1beta3 --template="{{ .spec.portalIP }}" service openldap-server)
 
-
 oc login -u system:admin -n default
 
+# readiness check is somehow faulty
+sleep 10
+
 echo "[INFO] Running extended tests"
 
-sleep 10
+# create a temp copy of the sync-configs and update them with the LDAP server's IP
+for config in test/extended/authentication/sync-config*.yaml
+do
+	cp ${config} ${BASETMPDIR}
+	os::util::sed "s/LDAP_SERVICE_IP/${LDAP_SERVICE_IP}/g" ${BASETMPDIR}/$(basename ${config})
+done
+
+# sync all groups from LDAP using rfc2307 schema, expect failure to find Person0
+! openshift ex sync-groups --sync-config=${BASETMPDIR}/sync-config-rfc2307.yaml --confirm 
+touch test/extended/authentication/valid_all_ldap_rfc2307_sync.txt
+oc get groups --no-headers | awk '{print $1}' | xargs -I{} oc get group {} -o yaml >> test/extended/authentication/valid_all_ldap_rfc2307_sync.txt
+oc delete groups --all
 
-cp test/extended/authentication/sync-schema1-group1.yaml ${BASETMPDIR}
-os::util::sed "s/LDAP_SERVICE_IP/${LDAP_SERVICE_IP}/g" ${BASETMPDIR}/sync-schema1-group1.yaml
-openshift ex sync-groups --sync-config=${BASETMPDIR}/sync-schema1-group1.yaml --confirm
+# sync all groups from LDAP using ad schema, 
+openshift ex sync-groups --sync-config=${BASETMPDIR}/sync-config-ad.yaml --confirm
+touch test/extended/authentication/valid_all_ldap_ad_sync.txt
+oc get groups --no-headers | awk '{print $1}' | xargs -I{} oc get group {} -o yaml >> test/extended/authentication/valid_all_ldap_ad_sync.txt
```

Comment:
`oc export groups -o yaml` will strip out generated fields making it better for this purpose than `oc get group`.  You can still pipe ordered names to it if you like:
```bash
oc get groups --no-headers -o names | sort | xargs oc export
```